server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java

```diff
@@ -106,7 +106,11 @@ public static XContentParser createParser(NamedXContentRegistry xContentRegistry
                 input = bytes.streamInput();
             }
             contentType = xContentType != null ? xContentType : XContentFactory.xContentType(input);
-            return new Tuple<>(Objects.requireNonNull(contentType), convertToMap(XContentFactory.xContent(contentType), input, ordered));
+            try {
```

Comment:
why does try/with not work here?

Reply:
It should, I just need to re-assign the `input` stream to a different var so that var can be closed. I'll do that